id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/helper/IdRepoServiceHelper.java

```diff
@@ -188,6 +188,15 @@ public Map<String, List<HandleDto>> getSelectedHandles(Object identity, Map<Stri
 		return null;
 	}
 
+    public Map<String,Object> updateSelectedHandleFields(Object identityObject, Map<String, List<HandleDto>> inputSelectedHandlesMap) throws IdRepoAppException {
+            String selectedHandlesFieldId = identityMapping.getIdentity().getSelectedHandles().getValue();
+            Map<String,Object> identityMap=convertToMap(identityObject);
+            if(identityMap.containsKey(selectedHandlesFieldId)){
+                List<String> selectedhandles = new ArrayList<String>(inputSelectedHandlesMap.keySet());
+                identityMap.put(selectedHandlesFieldId,selectedhandles);
+            }
+            return identityMap;
+    }
```

Comment:
Why is this method required, @dhanendra06? The same operation can be performed with the replaceConfiguredFieldsOnUpdate(inputData, dbData) method in the same class. Also, on update, "selectedHandles" should directly replace the old value in the database, regardless of whether the value existed previously.

Reply:
yes this method is required what if given input handle is failed or not valid then same selected handle field will be updated even handle is not stored for that we are checking the selected handle then based on that we are updating selected handle field